### PR TITLE
Fix Incorrectly Closing Connection when `stream.next()` Returns an Error

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

$Subject

Fixes: [#3565](https://github.com/ballerina-platform/ballerina-standard-library/issues/3565)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: ballerina/websocket_utils.bal [hunk 1]
```diff
@@ -86,7 +87,7 @@ isolated function getSubscriptionResponse(Engine engine, __Schema schema, Contex
     }
     string errorMessage = result is error ? result.message() : "Error ocurred in the subscription resolver";
     return {errors: [{message: errorMessage}]};
-    
+
```
Comment (DimuthuMadushan) [inline ballerina/websocket_utils.bal:90]: Extra new line?

File: ballerina/websocket_service.bal [hunk 2]
```diff
@@ -54,67 +54,80 @@ isolated service class WsService {
...
-            check sendWebSocketResponse(caller, self.customHeaders, WS_COMPLETE, null, connectionId);
-            closeConnection(caller);
-        } else if wsType == WS_PING {
+            WS_PING => {
             check caller->writeMessage({"type": WS_PONG});
-        } else if wsType == WS_PONG {
-            check caller->writeMessage({"type": WS_PING});
+            }
+            WS_PONG => {
+                check caller->writeMessage({"type": WS_PING});
```
Comment (MohamedSabthar) [inline ballerina/websocket_service.bal:111]: we don't have to write a ping message here

File: ballerina/websocket_utils.bal [hunk 3]
```diff
@@ -34,17 +40,12 @@ isolated function executeOperation(Engine engine, Context context, readonly & __
...
+            check sendWebSocketResponse(caller, customHeaders, WS_COMPLETE, null, connectionId);
         } else {
-            if customHeaders.hasKey(WS_SUB_PROTOCOL) {
-                check sendWebSocketResponse(caller, customHeaders, WS_COMPLETE, null, connectionId);
-            } else {
-                closeConnection(caller);
-            }
+            closeConnection(caller);
         }
+
```
Comment (DimuthuMadushan) [inline ballerina/websocket_utils.bal:48]: Extra new line?

File: ballerina/websocket_service.bal (hunk 2 above)
Comment (ThisaruGuruge) [reply ballerina/websocket_service.bal:111]: Did not do that change since you are working on the subprotocol fix.

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**